src/main/java/core/basesyntax/dao/impl/FruitStorageDaoImpl.java

```diff
@@ -0,0 +1,31 @@
+package core.basesyntax.dao.impl;
+
+import core.basesyntax.dao.FruitStorageDao;
+import core.basesyntax.db.FruitStorage;
+import core.basesyntax.model.FruitTransaction;
+import java.util.HashMap;
+import java.util.Map;
+
+public class FruitStorageDaoImpl implements FruitStorageDao {
+    @Override
+    public void addFruit(FruitTransaction fruitTransaction) {
+        FruitStorage.fruitStorage.put(fruitTransaction.getFruit(), fruitTransaction.getQuantity());
+    }
+
+    @Override
+    public void increaseQuantity(FruitTransaction fruitTransaction) {
+        FruitStorage.fruitStorage.merge(fruitTransaction.getFruit(),
+                fruitTransaction.getQuantity(), Integer::sum);
+    }
+
+    @Override
+    public void decreaseQuantity(FruitTransaction fruitTransaction) {
+        FruitStorage.fruitStorage.merge(fruitTransaction.getFruit(),
+                -fruitTransaction.getQuantity(), Integer::sum);
+    }
+
+    @Override
+    public Map<String, Integer> getAllAsMap() {
+        return new HashMap<>(FruitStorage.fruitStorage);
+    }
+}
```

Comment:
This class directly manipulates the `FruitStorage.fruitStorage` map. It would be better to encapsulate the access to the storage map within the `FruitStorage` class itself. Consider adding methods to the `FruitStorage` class that handle the addition, increase, and decrease of quantities, and then call those methods from here. This way, you adhere to the encapsulation principle and make the system more maintainable and less prone to errors.